lib/y2_installbase.pm

```diff
@@ -255,6 +258,55 @@ sub select_all_patterns_by_menu {
     assert_screen 'inst-overview';
 }
 
+=head2 select_not_install_any_pattern 
+
+    select_not_install_any_pattern() 
+
+Being in the "select pattern" screen, performs steps to not install any 
+patterns.
+=cut
+
+sub select_not_install_any_pattern {
+    my ($self) = @_;
+
+    # Ensure mouse on certain pattern then right click
+    assert_and_click("minimal-system", button => 'right');
+    assert_screen 'selection-menu';
+    # select action on all patterns
+    wait_screen_change { send_key 'a'; };
+    # confirm do not install
+    assert_and_click 'all-do-not-install';
+    save_screenshot;
+}
+
+=head2 select_current_visible_patterns 
+
+    select_current_visible_patterns()
+
+Being in the "select pattern" screen, performs steps to select visible
+patterns.
+=cut
+
+sub select_current_visible_patterns {
+    my ($self) = @_;
+    # Remove default patterns first
+    select_not_install_any_pattern();
```

Comment:
I don't think we need to couple with this code here, you can put outside in the new module suggested.

For this method we just need to create a new needle with [this](https://openqa.suse.de/tests/9680761#step/select_patterns/17) and [this](https://openqa.suse.de/tests/9680761#step/select_patterns/22) and do `assert_and_click`.
The tag for the needle shouldn't be `on-pattern` we need something different, like `pattern-$pattern`
Makes sense? Because this method looks complicated clicking always on home, going to the first item, which is not the idea, user could navigate to the end and select there two visible patterns. Besides I would use the existing iterative approach `process_patterns` for this instead. So my proposal is to just click both items directly with new code.

Reply:
As at on-pattern, it already use 'pattern-$pattern' tag, please check: https://openqa.suse.de/tests/9696311#step/select_patterns/16 tag: pattern-base.   To separate my assert_and_click pattern from on-pattern. I created tag named '$pattern-pattern'